hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4145,6 +4161,21 @@ func (r *HostedClusterReconciler) defaultAPIPortIfNeeded(ctx context.Context, hc
 	return nil
 }
 
+func (r *HostedClusterReconciler) defaultIngressDomain(ctx context.Context) (string, error) {
```

Comment:
why is this needed? Isn't this available at the HCP API, e.g `HCP.DNSSpec`?

Reply:
This is the ingress domain of the management cluster, not of the HC. We can obtain it from the ingress.config.openshift.io/cluster resource.

Reply:
Yeh I think it makes sense for the HC to fetch it as in the PR. But I think it should be passed as API input to the HCP as everything else, why is it using an env variable instead?

Reply:
The reason is that this is an operator-wide setting and nothing specific to a particular HostedCluster. It seemed wrong to pass it as a HostedControlPlane field. If/when it changes, we need to modify all existing HCPs instead of just redeploying the control plane operator.

Reply:
ok I see, why is it not a flag then as everything else? to clarify I have no objections to proceed with the env, just trying to understand why we are making a difference here.

Reply:
Because flags introduce compatibility problems that are avoided by env vars. We don't have a good way from the HO to determine whether a flag applies to a particular control plane binary. If we agree, I would prefer to use env vars for future params passed from HO to CPO. If the particular CPO doesn't support it, then it can just ignore it.

Reply:
@csrwng Make sense but then for long term sustainability it'd be good to try to consolidate and migrate all flags to be envs before cutting a release.